l1-contracts/test/fees/MinimalFeeModel.sol

```diff
@@ -38,30 +30,89 @@ contract MinimalFeeModel is TimeFns {
   Timestamp public immutable GENESIS_TIMESTAMP;
 
   uint256 public populatedThrough = 0;
-  mapping(uint256 _slotNumber => DataPoint _dataPoint) public dataPoints;
+  mapping(uint256 slotNumber => FeeHeader feeHeader) public feeHeaders;
 
-  L1BaseFees public l1BaseFees;
+  L1GasOracleValues public l1BaseFees;
 
   constructor(uint256 _slotDuration, uint256 _epochDuration) TimeFns(_slotDuration, _epochDuration) {
     GENESIS_TIMESTAMP = Timestamp.wrap(block.timestamp);
-    dataPoints[0] = DataPoint({provingCostNumerator: 0, feeAssetPriceNumerator: 0});
+    feeHeaders[0] = FeeHeader({
+      excess_mana: 0,
+      fee_asset_price_numerator: 0,
+      mana_used: 0,
+      proving_cost_per_mana_numerator: 0
+    });
+
+    l1BaseFees.pre = L1Fees({base_fee: 1 gwei, blob_fee: 1});
+    l1BaseFees.post = L1Fees({base_fee: block.basefee, blob_fee: _getBlobBaseFee()});
+    l1BaseFees.slot_of_change = LIFETIME.unwrap();
+  }
+
+  function getL1GasOracleValues() public view returns (L1GasOracleValues memory) {
+    return l1BaseFees;
+  }
 
-    l1BaseFees.pre = BaseFees({baseFee: 1 gwei, blobFee: 1});
-    l1BaseFees.post = BaseFees({baseFee: block.basefee, blobFee: _getBlobBaseFee()});
-    l1BaseFees.slotOfChange = LIFETIME;
+  // For all of the estimations we have been using `3` blobs.
+  function manaBaseFeeComponents(uint256 _blobsUsed, bool _inFeeAsset)
+    public
+    view
+    returns (ManaBaseFeeComponents memory)
+  {
+    L1Fees memory fees = getCurrentL1Fees();
+    uint256 dataCost =
+      Math.mulDiv(_blobsUsed * 2 ** 17, fees.blob_fee, FeeMath.MANA_TARGET, Math.Rounding.Ceil);
+    uint256 casUsed = FeeMath.L1_GAS_PER_BLOCK_PROPOSED + _blobsUsed * 50_000
```

Comment:
```suggestion
    uint256 gasUsed = FeeMath.L1_GAS_PER_BLOCK_PROPOSED + _blobsUsed * 50_000
```

Reply:
the 50 comes from a const?

Reply:
The cost of `0x0a` pre-compile. Most of these numbers atm are pretty much pulled out of the ass though.